openapi.yml

```diff
@@ -37,8 +37,14 @@ paths:
                       type: object
                       properties:
                         dl_vlan:
-                          type: integer
                           description: VLAN ID
+                          oneOf:
+                            - type: integer
+                              format: int32
+                            - type: string
+                              enum:
+                                - any
+                                - untagged                          
```

Comment:
Regarding the API body input for `dl_vlan` I think it shouldn't change. Although you still need to be able to trace `dl_vlan: 0` and `dl_vlan: "4096/4096"`, the input of this API, as far as I can tell, is meant to emulate the fields of a packet following the name convention of OpenFlow but without supporting wildcard (mask) in the input, so supporting wildcard in the input would start to diverge from this, could we support also a wildcard in the input? Theoretically, we could, so far I don't think we have requirements for this though, and for [vlan range the discussion](https://github.com/kytos-ng/kytos/issues/318) is also considering to continue to use the API without any wildcards, so I'll recommend to stick with that just so it continues to be aligned.

Reply:
Also having `"eth"` with `"untagged"` in a `"dl_vlan"` since `"eth"` is already optional it would be two forms of representing the same concept. An "untagged" packet doesn't have a VLAN so you omit `"dl_vlan"` it's supposed to be actual untagged, so this is also another indication that the API would become confusing.

Reply:
This makes so much sense.